cl/citations/api_views.py

```diff
@@ -0,0 +1,209 @@
+from http import HTTPStatus
+
+import eyecite
+from asgiref.sync import async_to_sync
+from django.db.models import QuerySet
+from django.http import HttpResponse
+from django.template.defaultfilters import slugify
+from reporters_db import EDITIONS
+from rest_framework.exceptions import NotFound
+from rest_framework.generics import GenericAPIView
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from cl.api.pagination import MediumAdjustablePagination
+from cl.citations.api_serializers import CitationRequestSerializer
+from cl.citations.exceptions import MultipleChoices
+from cl.citations.utils import get_canonicals_from_reporter
+from cl.search.api_serializers import OpinionClusterSerializer
+from cl.search.models import OpinionCluster
+from cl.search.selectors import get_clusters_from_citation_str
+
+SLUGIFIED_EDITIONS: dict[str, str] = {
+    str(slugify(item)): item for item in EDITIONS.keys()
+}
+
+
+class CitationLookupView(GenericAPIView):
+    queryset = OpinionCluster.objects.all()
+    pagination_class = MediumAdjustablePagination
+    serializer = OpinionClusterSerializer
+
+    def get(self, request: Request, *args, **kwargs) -> HttpResponse:
+        query = request.query_params
+
+        # Uses the serializer to perform object level validations
+        citation_serializer = CitationRequestSerializer(data=query)
+        citation_serializer.is_valid(raise_exception=True)
+
+        # Get query parameters from the validated data
+        data = citation_serializer.validated_data
+        self.full_text_citation = data.get("text_citation", None)
+        self.reporter = data.get("reporter", None)
+        self.volume = data.get("volume", None)
+        self.page = data.get("citation_page", None)
+
+        if self.full_text_citation:
+            citations = eyecite.get_citations(self.full_text_citation)
+            if not citations:
+                raise NotFound(
+                    f"Unable to find a citation withing the provided text"
+                )
+
+            if citations[0].groups:
+                c = citations[0]
+                self.reporter = slugify(c.groups["reporter"])
+                self.volume = c.groups["volume"]
+                self.page = c.groups["page"]
+            else:
+                raise NotFound(
+                    f"The provided text is not a valid citation. Please review it and try again"
```

Comment:
Should this be a different status code for invalid input, say?

Also:
```suggestion
                    f"Invalid citation."
```

Reply:
> Should this be a different status code for invalid input

You're correct. We should use status code 400 becuase the request has malformed data